.circleci/config.yml

```diff
@@ -76,6 +78,12 @@ jobs:
       - run: date +%F > date
       - restore_cache:
           <<: *restore_cache
+      - run:
+          name: CodeClimate before-build
```

Comment:
Why do we need to do this before running the tests? Can we follow the same layout as that for codecov where we install and generate the coverage report in the same step?
Also, please use descriptive names for the steps.

Reply:
done